fs-objects.go

```diff
@@ -143,7 +145,73 @@ func (fs fsObjects) GetObjectInfo(bucket, object string) (ObjectInfo, error) {
 
 // PutObject - create an object.
 func (fs fsObjects) PutObject(bucket string, object string, size int64, data io.Reader, metadata map[string]string) (string, error) {
-	return putObjectCommon(fs.storage, bucket, object, size, data, metadata)
+	// Verify if bucket is valid.
+	if !IsValidBucketName(bucket) {
+		return "", BucketNameInvalid{Bucket: bucket}
+	}
+	// Check whether the bucket exists.
+	if !isBucketExist(fs.storage, bucket) {
+		return "", BucketNotFound{Bucket: bucket}
+	}
+	if !IsValidObjectName(object) {
+		return "", ObjectNameInvalid{
+			Bucket: bucket,
+			Object: object,
+		}
+	}
+
+	fileWriter, err := fs.storage.CreateFile(bucket, object)
```

Comment:
As discussed on phone having it in `tmp` and then renaming will have race condition, we have to bring in a fix where object in tmp is created as this path `.minio/tmp/UUID/path/to/object` but i will make the change to write it to `tmp` for now

but the separation was not driven by this but more because we have to do this for XL:

```
+	// check if an object is present as one of the parent dir.
 +	if err = xl.parentDirIsObject(bucket, path.Dir(object)); err != nil {
 +		return "", toObjectErr(err, bucket, object)
 +	}
```